### DISCUSSION TITLE
Link in the sidebar to go to the edit page

### DISCUSSION THREAD

**Discussion by MWA-NASMO** [filamentphp/filament#6116] [Help]
Hello,
I have created a resource for a Model wich has only one Entry in the data base an I only want an Edit view (no list, no create).
How can I do to make the link that is in the navigation sidebar go directly to the Edit page of that record (ID:1) ie. replace the default index by the Edit Page.

Thanks for your help.

**Comment by danharrin** [comment]
That's not a resource anymore. Resources need a List page at least.

Instead of a resource, create a custom page & use the form builder to render a form and save the contents to the table.

**Reply by MWA-NASMO** [reply]
Hello, I'm sorry but I can't find the good way to do that, If i create a custom page, it creates a view page. 
What I want is to use the EditRecord Class :
<?php

namespace App\Filament\Resources\ModeleLCResource\Pages;

use App\Filament\Resources\ModeleLCResource;
use Filament\Resources\Pages\EditRecord;

class ModeleLC extends EditRecord
{
    protected static string $resource = ModeleLCResource::class;

    protected function getActions(): array
    {
        return [
            //Actions\DeleteAction::make(),
        ];
    }
}
My problem is to specify. the record to be edited (ID:1)

**Reply by MWA-NASMO** [reply]
Extra question : wold you have links to a good training, On youtube, they all talk about basic functionalities like blogs, posts... nothing really useful

**Reply by danharrin** [reply]
A custom page doesnt have to belong to a resource. They are standalone and registered in the sidebar.

**Reply by MWA-NASMO** [reply]
So, the. form builder is set in the view page or in the filament page ? Do I have to write the form myself like in old Laravel projects. ?
There is not a lot of examples to how to do that...

**Reply by danharrin** [reply]
Create a custom page, which is a Livewire component internally. Then look at the "Getting Started" docs for the Form Builder, which shows you how to add a form to the page.

**Reply by MWA-NASMO** [reply]
Do I have. to istall. filament-form even if i Use filament Admin-panel ? I thought all packages were already installed.

**Reply by danharrin** [reply]
It is already installed

**Reply by MWA-NASMO** [reply]
Thank you, i really appreciate your help !
Do I have to create a livewire Component or can I set my form in my Resource/Pages/ file ? and how do I specify that it is the record ID=1 that i want to edit. do I have to declare public $record = Model::first(); to pass it to my form ?
I'm really lost and find no one to help me except you...

**Reply by danharrin** [reply]
`php artisan make:filament-page PageName`. It does not need to belong to a resource

You do not need an extra Livewire component. That is the Livewire component. You can use the form builder directly on it.

You do not need to declare `$record` anywhere if there is only 1 record you want to edit. You will need to learn some Livewire and read our docs about what to put in `mount()`, `getFormSchema()` and your submit/save function.

**Reply by michaelklopf** [reply]
I'm sorry, but I would like to add that if anyone wants to create a page with an infolist, it's similar. The page is a Livewire component so you can follow the docs at [Adding an infolist to a Livewire component](https://filamentphp.com/docs/3.x/infolists/adding-an-infolist-to-a-livewire-component) to add the infolist to the page. I added the `mount()` method to load my object from the database that is used as record in the infolist.

**Comment by steveandsten** [comment]
You could simply add the following to your Resource File...

protected static bool $shouldRegisterNavigation = false;

Then create your own navigational item in the AdminPanelProvider.php

->navigationItems([
    NavigationItem::make('Settings')
        ->url('/admin/settings/1/edit')
        ->icon('heroicon-o-cog-6-tooth')
        ->sort(100)
        ->group('System Management')
        ->visible(fn (): bool => auth()->user()?->user_type === 'admin'),
])

It's not the prettiest, but will work
